SERVER-98272 Create remediation for if the timestamp embedded in the bucket's ID does not match the control.min timestamp (#144)
mongodb/support-tools: embedded-timestamp-mismatch/README.md

                                  "bucket '_id' field and the timestamp [...] in 'control.min' field."}}...
```

# Remediation

## Rewrite the Buckets with Mismatched Embedded Bucket ID Timestamp and control.min Timestamp

While the script is running, the performance of operations on the time-series collection may be impacted. The script does a scan of the whole collection and performs updates on impacted buckets, which may result in a large load if many buckets are affected.  

At a high level, the script remediates errors during collection validation by rewriting buckets that are detected to have a mismatched embedded bucket ID timestamp and control.min timestamp. The rewrite is done by unpacking the measurements of the problematic buckets and inserting those measurements back into the collection.

The full steps are as follows. For each bucket in the time series collection:
- Detect if the bucket has mismatched embedded bucket ID timestamp and control.min timestamp
- Re-insert the measurements of the problematic bucket transactionally.
  - Unpack the measurements.

>   - Insert the measurements back into the collection. These will go into a new buckets.
>   - Delete the problematic bucket from the collection.

Reviewer: [nit] Is it clearer if it's worded as
```suggestion
  - Repack the measurements into new buckets.
  - Delete the original, problematic bucket from the collection.
```
?
